features/FEATURE_LWIP/lwip-interface/lwip_stack.c

```diff
@@ -84,6 +116,18 @@ static struct lwip_socket *mbed_lwip_arena_alloc(void)
 static void mbed_lwip_arena_dealloc(struct lwip_socket *s)
 {
     s->in_use = false;
+
+    while(s->multicast_memberships_count > 0) {
+        static uint32_t index = 0;
+        index = next_registered_multicast_member(s, index);
+
+        mbed_lwip_setsockopt(0, s, 0, NSAPI_DROP_MEMBERSHIP, &s->multicast_memberships[index],
```

Comment:
Third parameter should technically be `NSAPI_SOCKET` - at some point `mbed_lwip_setsockopt` should enforce that.

First parameter should be written as `NULL`.